### ISSUE TITLE
False positive warning for CA1508 ("Avoid dead conditional code")

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#55223]
**Version Used**: 
dotnet 5.0.301
vscode

**Steps to Reproduce**:

I receive an input that is allowed to be null, but is not allowed to be empty nor whitespace.

````cs
var s = "  \t";
if (s != null && string.IsNullOrWhitespace(s)) {    // <-- warning CA1508
  // ...
}
````

Warning:
> 'string.IsNullOrWhiteSpace(s)' is always 'false'. Remove or refactor the condition(s) to avoid dead code.

**Expected Behavior**:
If it is known that the string is not null, then the "empty or whitespace" behavior of `IsNullOrWhiteSpace` still applies. The warning should not occur.

**Comment by lonix1** [comment]
Sorry didn't see there was a template for analyzer bugs. Can this be moved to that repo?

**Comment by Youssef1313** [comment]
@jinujoseph or @mavasani to move to roslyn-analyzers.
